configure.ac

```diff
@@ -2884,15 +2884,26 @@ AS_IF([test "$with_dln_a_out" != yes], [
   AS_IF([test "$rb_cv_dlopen" = yes], [
     AS_CASE(["$target_os"],
       [darwin*], [
+        AC_SUBST(ADDITIONAL_DLDFLAGS, "")
 	for flag in \
-	  "-undefined dynamic_lookup" \
 	  "-multiply_defined suppress" \
+	  "-undefined dynamic_lookup" \
 	  ; do
-	  test "x${linker_flag}" = x || flag="${linker_flag}`echo ${flag} | tr ' ' ,`"
-	  RUBY_TRY_LDFLAGS([$flag], [], [flag=])
-	  AS_IF([test "x$flag" != x], [
-	    RUBY_APPEND_OPTIONS(DLDFLAGS, [$flag])
-	  ])
+      test "x${linker_flag}" = x || flag="${linker_flag}`echo ${flag} | tr ' ' ,`"
+      RUBY_TRY_LDFLAGS([$flag], [], [$flag=])
```

Comment:
@hsbt Should this not be `RUBY_TRY_LDFLAGS([$flag], [], [flag=])`? The last expression, `[flag=]` not `[$flag=]`?

Reply:
https://github.com/ruby/ruby/blob/master/configure.ac#L3110